readme.md



## Suggested Ways to Get Started

- Break the project down into different components (data, presentation, views, style, DOM manipulation) and brainstorm each component individually. Use whiteboards!
- Use your Development Tools (console.log, inspector, alert statements, etc) to debug and solve problems
- Work through the lessons in class & ask questions when you need to! Think about adding relevant code to your game each night, instead of, you know... _procrastinating_.
- Commit early, commit often. Don’t be afraid to break something because you can always go back in time to a previous version so long as it's been saved as a commit.
- Consult documentation resources (MDN, CSS Tricks etc.) at home to better understand what you’ll be getting into.
- Don’t be afraid to write code that you know you will have to remove later. Create temporary elements (buttons, links, etc) that trigger events if real data is not available. For example, if you’re trying to figure out how to change some text when the game is over but you haven’t solved the win/lose game logic, you can create a button to simulate that until then.



monte-game/changes.md

// logic
// three cards
// array of 3 objects
//2- shuffle my array
//3- check if  click was the correct guess
//4 -win message
//5 - let player1Card/player2Card.
//write announcecard function
// to check who won the game, we must compare player card
//post - add animated progress bar.
//post change alert color
// dont forget to load back the prompt.


// BETTING PROMPT
// let userMoney = 200; //User starts with this amount
// alert(`You have a total of $${userMoney} right now`);
//
// //How much the user is betting
// let userBet = window.prompt("How much do you want to bet?");
//
// // If statement checks if the user bet the right amount
// if (Number(userBet) > userMoney) {
//   userBet = window.prompt("You don't have enough money, try again");
// } else if (Number(userBet) < 50){
//   userBet = window.prompt("Please bet $50 or higher");
// }
//
// // Converts bet from string to number type
// let gamePot = Number(userBet);
//
// // Subtracts the bet from user money
// userMoney -= gamePot;
//
// gamePot += gamePot; //doubles gamepot
// alert(`The pot is $${gamePot}`);
// alert(`You have a total of $${userMoney} remaining`);
//
// // Gets the elements that hold the userMoney and gamePot ID inside the HTML
// let money = document.getElementById('userMoney');
// let pot = document.getElementById('gamePot');
//
// // Adds following money amount to the html elements
// money.innerHTML = `User money ${userMoney}`;
// pot.innerHTML = `Game pot ${gamePot}`;
//******* ends here***
// creating random index.
// const randomSelect= Math.floor(Math.random() * cards.length);
// console.log(cards[randomSelect]); // picks random card.



const cards = [
  {card: 'Ace', suit: 'Spades'},
  {card: 'Ace', suit: 'Hearts'},
  {card: 'Ace', suit: 'Diamonds'},
];


let containerDiv = document.querySelector('.container');

function dynamicallyCreateCards() {
  for (let i = 0; i < cards.length; i ++) {
    let cardObj = document.createElement('div');
    cardObj.classList.add('card');
    cardObj.innerHTML = `${cards[i].card} of ${cards[i].suit}`;

    cardObj.addEventListener("click", (evt) => {
      console.log(`${cards[i].card} of ${cards[i].suit}`);
    }) // eventlistener.

    containerDiv.appendChild(cardObj);
    setTimeout(function () { window.scrollTo(0,1)},1);

    // setTimeout(function () {
    //   $(".card").rotate(true);
    // }, 5000);

    // document.querySelector(".card").classList.toggle("flip"); // example from david walsh css flip
  }
  // dynamicallyCreateCards();
}
dynamicallyCreateCards();

// This shuffle algorithm was provided by Kenny's WDI in-class lab
function shuffleCards(deck) {
  return deck.sort(() => 0.5 - Math.random());
}

// 1. Add click event listeners to each card


// 2. Display the card's details when I click it
// 3. Shuffle my array of card objects
// 4. How do I shuffle the cards before they get dynamically created
// const randomSelect= math.floor(math.Random() * deck.length);
// 5. Make a message popup showing the card's details when clicked
// onst cardFlipped
// button on click should start game.



//// 12/18/19- bug ////
// let images =[]; // all svg images for game will be in this array.
// images. slice and concat(images)
let gameCard = document.getElementById("game-card");
let gameBox = document.getElementById("game");
let gameclick = document.getElementById("btn")
let containerDiv = document.querySelector('.container');

function dynamicallyCreateCards() {
  for (let i = 0; i < cards.length; i ++) {
    let cardObj = document.createElement('div');
    cardObj.classList.add('card');
    cardObj.innerHTML = `${cards[i].card} of ${cards[i].suit}`;

    cardObj.addEventListener("click", (evt) => {
      console.log(`${cards[i].card} of ${cards[i].suit}`);
    }) // eventlistener.

    containerDiv.appendChild(cardObj);
    setTimeout(function () { window.scrollTo(0,1)},1);


}
dynamicallyCreateCards();



/////



// 3 EASY STEPS FOR DYNAMICALLY CREATING ELEMENT WITH JS
// let containerDiv = document.querySelector('#container');1
// let card = document.createElement('div');2
// containerDiv.appendChild(card);3
//create element, add clas to element, append the element to body.



1- cards pics from vecteezy.com
reference
https://css-tricks.com/


monte-game/main.js

// logic
// three cards
// array of 3 objects
//2- shuffle my array
//3- check if  click was the correct guess
//4 -win message
//5 - let player1Card/player2Card.
//write announcecard function
// to check who won the game, we must compare player card
//post - add animated progress bar.
//post change alert color
// dont forget to load back the prompt.

//
// let cardImages = [
// 'back1.png',
// 'diamond1.png',
// 'heart1.png',
// 'spades1.png'
// ];

var mySound;
var audioMp3 = new Audio();



const cards = [
  {card: 'Ace', suit: 'Spades'},
  {card: 'Ace', suit: 'Hearts'},
  {card: 'Ace', suit: 'Diamonds'},
];

// shuffleCards.shuffle();
function shuffleCards(deck) {                          //2 created a function that takes deck aas parameter, randomize and sorts it.
  return deck.sort(() => 0.5 - Math.random());
}


let containerDiv = document.querySelector('.monte');
let shuffledDeck = shuffleCards(cards);   // assigned a callback function to variable, so we can use in a for loop.



function createCards() {   // create card funciton, which is loping, then a div, and adding a class list, .
  for (let i = 0; i < shuffledDeck.length; i ++) {


    let cardObj = document.createElement('div');
    cardObj.classList.add('card');
    console.log(cardObj);
    cardObj.setAttribute('data-name', `${shuffledDeck[i].suit}`)    /// added data-name so we can use it to comare the cards. string intepulation, attiching key value pair.

// cardObj


if (shuffledDeck[i].suit === 'Spades') {         // assisng css class
  cardObj.classList.add('spades');
} else if (shuffledDeck[i].suit === 'Hearts') {
  cardObj.classList.add('heart');
  console.log('this worked')
} else {
  cardObj.classList.add('diamond');
}

    cardObj.addEventListener("click", (e) => {
      // evt.preventDefault();

      console.log(`${shuffledDeck[i].card} of ${shuffledDeck[i].suit}`);
      e.target.classList.remove('shuffle-cards');
      e.target.classList.add('show-card');


      // console.log(cardObj);
      // check to see if when card is clicked the flip class is added.
      // check if what you clicked matches ace of spade
      if (e.target.dataset.name === 'Spades') {
        console.log(e.target)
        alert("you win");
        // e.target.style.transform = 'rotateY(180deg)';
        // alert("you won!");
      }
      else{
        console.log("get lost");
      // cardObj.classList.add('hide-card');
      // console.log("get lost");
      }

    });

    containerDiv.appendChild(cardObj);
    // setTimeout(() => {
    //   // evt.preventDefault();  // prevents refreshing of page
    //   cardObj.classList.add('hide-card');
    // }, 1000);
    setTimeout(() => {
      cardObj.classList.add('shuffle-cards');

    }, 1000);

  }
}
createCards();
// mySound = new sound("gametheme.mp3");
// mySound.play();



function startGame() { // doesnt pop up immediately, except on click.
// // BETTING PROMPT


  let userMoney = 200; //User starts with this amount
  alert(`You have a total of $${userMoney} right now`);

  //How much the user is betting
  // trying let start game buttn prompt users.
  let userBet = window.prompt("How much do you want to bet?");

  // If statement checks if the user bet the right amount
  if (Number(userBet) > userMoney) {
    userBet = window.prompt("You don't have enough money, try again");
  } else if (Number(userBet) < 50){
    userBet = window.prompt("Please bet $50 or higher");
  }

  // Converts bet from string to number type
  let gamePot = Number(userBet);

  gamePot += gamePot; //doubles gamepot
  alert(`The pot is $${gamePot}`);
  alert(`You have a total of $${userMoney} remaining`);

  // Gets the elements that hold the userMoney and gamePot ID inside the HTML
  let money = document.getElementById('userMoney');
  let pot = document.getElementById('gamePot');

  // Adds following money amount to the html elements
  money.innerHTML = `User money ${userMoney}`;
  pot.innerHTML = `Game pot ${gamePot}`;

  // creating random index.
  // const randomSelect = Math.floor(Math.random() * cards.length);
  // console.log(cards[randomSelect]); // picks random card.
}

// let allCards = document.querySelectorAll(".card");
// console.log(allCards);


function playAudio(){
  // audioMp3.src = 'http://www.qingli.life/music/Beyond_Jupiter.mp3';
  audioMp3.src = 'assets/Malicious.mp3';

	audioMp3.play();
}


let replayButton = document.querySelector(".rePlay");
replayButton.addEventListener("click", function() {

  // location.reload();
  document.body.onload = createCards();

//   allCards.forEach((crd) => {
// console.log(card);
// card.style.transform = "rotateY(-180deg)";
// });
});
// audioMp3.play();
